src/javascript/EditPanel/EditPanelContent/FormBuilder/FieldSet/Field/Field.jsx

```diff
@@ -96,26 +103,35 @@ export const FieldCmp = ({classes, inputContext, idInput, selectorType, field, f
         }
 
         onChangeValue.current = currentValue;
-    };
+    }, [field, registeredOnChanges, selectorType]);
 
-    const onChange = currentValue => {
+    const onChange = useCallback(currentValue => {
         // Update formik
         setFieldValue(field.name, currentValue, true);
         // Validation has been done on the setValue, no need to redo it on touch.
         setFieldTouched(field.name, field.isMultiple ? [true] : true, false);
 
         // Trigger on changes
         registeredOnChange(currentValue);
-    };
+    }, [field.isMultiple, field.name, registeredOnChange, setFieldTouched, setFieldValue]);
+
+    const registeredOnChangeRef = useRef(registeredOnChange);
+    useEffect(() => {
+        registeredOnChangeRef.current = registeredOnChange;
+    }, [registeredOnChange]);
+
+    const initialValue = useRef(values[field.name]);
 
     useEffect(() => {
```

Comment:
This effect should only be called on initial mount, with initial value , and complete field unmount ( so `[]` deps ), but we still need to call the last version of registeredOnChange (which can be updated when field change) -> use a ref.
Also use a ref to get the initial value (ref not updated afterward)